tasks/main.yml

```diff
@@ -1,36 +1,107 @@
+  become_user: "{{ postgresql_user }}"
+
+- name: Ensure postgresql "role" for backups has "select" access to any table in {{ backups_role_db_name }}
+  postgresql_privs:
+    db: "{{ backups_role_db_name }}"
+    role: "{{ backups_role_postgresql_user_name }}"
+    objs: ALL_IN_SCHEMA
+    # This ensures that this priv exists: {{ role }}=r/{{ become_user }}
+    privs: SELECT
+  become: true
+  become_user: "{{ postgresql_user }}"
+
+- name: Ensure postgresql "role" for backups has no other access to any table in {{ backups_role_db_name }}
+  postgresql_privs:
+    db: "{{ backups_role_db_name }}"
+    role: "{{ backups_role_postgresql_user_name }}"
+    objs: ALL_IN_SCHEMA
+    # Revoke all privs except SELECT
+    # https://www.postgresql.org/docs/current/sql-grant.html
+    privs: "INSERT,UPDATE,DELETE,TRUNCATE,REFERENCES,TRIGGER"
+    state: absent
+  become: true
+  become_user: "{{ postgresql_user }}"
+
+- name: Let backup user to use `{{ backups_role_sudoers_cmd_pattern }}` as root
+  lineinfile:
+    path: /etc/sudoers
+    state: present
+    # Pattern to match the line
+    regexp: '^{{ backups_role_user_name }} ALL=(root)'
+    # Actual line to exist in the file
+    line: "{{ backups_role_user_name }} ALL=(root) NOPASSWD: {{ backups_role_sudoers_cmd_pattern }}"
+    validate: '/usr/sbin/visudo -cf %s'
+
 - name: Install restic and configure restic repository
   vars:
     restic_version: "{{ backups_role_restic_version }}"
     restic_repos:
       - name: "{{ backups_role_restic_repo_name }}"
-        url: "{{ backups_role_bucket_url }}"
-        password: "{{ backups_role_restic_password }}"
+        url: "{{ backups_role_restic_repo_url }}"
+        password: "{{ backups_role_restic_repo_password }}"
         remote_credentials:
-          b2_account_id: "{{ backups_role_b2_account_id }}"
-          b2_account_key: "{{ backups_role_b2_account_key }}"
-    jobs:
-      - command: "{{ backups_role_path }}/backups.sh && restic backup {{ backups_role_path }}/pg_dump {{ backups_role_path }}/config.tar.gz {{ backups_role_assets_paths|join(' ') }}"
-        at: "{{ backups_role_cron_job_at }}"
-        user: "{{ backups_role_user }}"
-      - command: 'restic forget --keep-last 5 --prune'
-        at: '5 11 */5 * *'
-        user: "{{ backups_role_user }}"
+          # See README.md for an explanation
+          # we cross app and account credentials here.
+          b2_account_id: "{{ backups_role_b2_app_key_id }}"
+          b2_account_key: "{{ backups_role_b2_app_key }}"
+        jobs:
+          - command: "{{ backups_role_script_path }}/{{ backups_role_script_name }}"
+            at: "{{ backups_role_cron_job_at }}"
+            user: "{{ backups_role_user_name }}"
+    restic_user: "{{ backups_role_user_name }}"
+    restic_group: "{{ backups_role_user_group }}"
+    restic_cron_stdout_file: "{{ backups_role_cron_stdout_file }}"
+    restic_cron_stderr_file: "{{ backups_role_cron_stderr_file }}"
+    
   include_role:
```

Comment:
This should go in the same block as line `85`, right?

Reply:
Yes. It confused me too. Adresses in f4c40f407ee581acbbf58da486a960f465fd72d9